### ISSUE TITLE
offen i have this error "Could not connect to {}".format(mac)"

### ISSUE THREAD

**Issue by uowis** [mjmccans/hassio-addon-airthings#21]
`[2022-08-28 18:33:49] DEBUG: Retrying d8:71:4d:96:c3:50

[2022-08-28 18:33:59] INFO: Not able to connect to d8:71:4d:96:c3:50

[2022-08-28 18:33:59] ERROR: Error getting device info for d8:71:4d:96:c3:50: Could not connect to d8:71:4d:96:c3:50

Traceback (most recent call last):
  File "/src/airthings.py", line 273, in get_info
    raise Exception("Could not connect to {}".format(mac))
Exception: Could not connect to d8:71:4d:96:c3:50

[2022-08-28 18:33:59] DEBUG: Disconnected.

[2022-08-28 18:33:59] DEBUG: Getting sensors...

[2022-08-28 18:33:59] DEBUG: Getting sensors for d8:71:4d:96:c3:50

[2022-08-28 18:33:59] DEBUG: Connecting to d8:71:4d:96:c3:50

[2022-08-28 18:34:09] DEBUG: Retrying d8:71:4d:96:c3:50`

**Comment by mjmccans** [comment]
Apologies for the delay responding. I was away for a while, but more importantly I am holding off making changes to this addon because it looks like support for Airthings devices is likely being brought into Home Assistant Core (see this [pull request](https://github.com/home-assistant/core/pull/77284)). I understand that it almost made it into 2022.09 and is currently slated to be included with 2022.10. The addition to Core includes the changes I made to use the new Bleak library, and is a better solution than this addon. For example, it will include device discovery, which is a much better user experience. Fingers crossed it will make it in and this addon will be rendered obsolete.

**Comment by TheQue42** [comment]
Okey, that sounds fine. Thanks for all the work you made to this addon anyway :-)

**Comment by dipcore** [comment]
But in fact that error appears for me too. I hope it is not related to Bleak, and we wont have the same error with Home Assistant Core

**Comment by dipcore** [comment]
I have a feeling this error happens when two or more devices used. Similar behavior I have for the component you mention (https://github.com/home-assistant/core/pull/77284 I installed it as a custom component to check it out)
```
2022-10-25 12:23:30.815 ERROR (MainThread) [custom_components.airthings_ble.config_flow] Error connecting to and getting data from E4:E1:12:29:78:59: E4:E1:12:29:78:59 - /org/bluez/hci0/dev_E4_E1_12_29_78_59: Failed to connect: [org.freedesktop.DBus.Error.UnknownObject] Method "Connect" with signature "" on interface "org.bluez.Device1" doesn't exist
``` 
So as a result one sensor is added and used properly. For second one I get an error above.

**Comment by mjmccans** [comment]
The error you are getting is more about the addon being overly verbose in its output. I am no bluetooth expert, but I have found connecting to devices to be a bit finicky, particularly when they are farther away. Therefore, often connections fail which is why there are a number of retries.

In other news, it looks like the new Airthings BLE will be included in 2022.11 (see https://rc.home-assistant.io/blog/2022/10/26/release-202211/#new-integrations). I have not used it yet but, as mentioned before, I believe native support will provide a better and more reliable experience than my addon.

**Comment by dipcore** [comment]
You are right. After abit more detailed investigation, I found out that is directly related to bluetooth connection and most likely interference with sonoff zigbee usb dongle. When I have it plugged directly to usb port of my RPi4 Airthings sensor located in my basement often gets disconnected.  And I have tons of retries. 

The gaps on the chart are connection issues I had:
![image](https://user-images.githubusercontent.com/2119471/198795970-cd1de874-849a-4943-99c3-0b202472117c.png)

NB Using usb extension cable and placing zigbee dongle 30 cm away seems solve the issue.

**Comment by uowis** [comment]
https://www.home-assistant.io/integrations/airthings_ble), added by [@vincegio](https://github.com/vincegio)](https://www.home-assistant.io/blog/2022/11/02/release-202211/)
I'm triying it

**Comment by dipcore** [comment]
@uowis I just updated and it works fine for me for two devices! (with using usb extension cable to move zigbee dongle away)

**Comment by uowis** [comment]
yes it make the Job but, it will be better that work with MQTT to keep value,..
